### PR TITLE
Fixed the ScatterGatherFirstCompleted router logic

### LINKED ISSUE #1718: Router with ScatterGatherFirstCompletedRoutingLogic replies with Task<object>
Hi

I think, there is a bug in ScatterGatherFirstCompletedRoutingLogic. Router with it replies with task (with message as result) instead of a message. 
Routers with TailChoppingRoutingLogic reply with normal message.

Test project:
[Program.cs.txt](https://github.com/akkadotnet/akka.net/files/137694/Program.cs.txt)

Output:

> Using ScatterGatherFirstCompletedRoutingLogic = False
> Result is Task'1
> Result is MessageB
> Using ScatterGatherFirstCompletedRoutingLogic = True
> Result is Task'1
> Result is Task'1
> Result is MessageB

If it is not a desired behaviour, maybe additional unwrapping will help at akka.net/src/core/Akka/Routing/ScatterGatherFirstCompleted.cs:87
Smth like: `var any =  Task.Run(async () => await await Task.WhenAny(tasks));`

Thanks

### PR DESCRIPTION
**Pull request by annymsMthd**
Fixes #1718  The spec was using broadcast which never goes through the specific router routee logic

### CODE REVIEW

File: src/core/Akka/Routing/ScatterGatherFirstCompleted.cs [hunk 1]
```diff
@@ -78,14 +79,18 @@ public ScatterGatherFirstCompletedRoutees(Routee[] routees, TimeSpan within)
         /// <param name="sender">The actor sending the message.</param>
         public override void Send(object message, IActorRef sender)
         {
```
Comment (annymsMthd) [inline src/core/Akka/Routing/ScatterGatherFirstCompleted.cs]: I'm a bit iffy on the error handling here. What happens if one of these tasks throws?

**Comment by Aaronontheweb** [comment]
@annymsMthd looks like there are public API changes in here - you'll need to follow this: http://getakka.net/docs/akka-developers/public-api-changes

**Comment by annymsMthd** [comment]
I have a few upgrades im doing for this

File: src/core/Akka/Routing/ScatterGatherFirstCompleted.cs (hunk 1 above)
Comment (annymsMthd) [inline src/core/Akka/Routing/ScatterGatherFirstCompleted.cs]: Fixed this by using task completion sources and Status.Failure messages

File: src/core/Akka.Tests/Routing/ScatterGatherFirstCompletedSpec.cs [hunk 2]
```diff
@@ -112,27 +159,41 @@ protected override void OnReceive(object message)
...
-            var routedActor = Sys.ActorOf(new ScatterGatherFirstCompletedGroup(paths, TimeSpan.FromSeconds(3)).Props());
+            private int _id;
+            private readonly TimeSpan _waitTime;
 
-            routedActor.Tell(new Broadcast(new Stop(1)));
-            shutdownLatch.Open();
-            var res = routedActor.Ask<int>(new Broadcast(0), TimeSpan.FromSeconds(10));
-            res.Wait();
-            res.Result.ShouldBe(14);
+            public DelayStopActor(int id, TimeSpan waitTime)
```
Comment (annymsMthd) [inline src/core/Akka.Tests/Routing/ScatterGatherFirstCompletedSpec.cs]: Prob don't need this. I saw that the other actor delays based on id so i can just use that functionality

**Pull request merged**